phoenix-core/src/it/java/org/apache/phoenix/end2end/CDCStream2IT.java

```diff
@@ -0,0 +1,174 @@
+        String tableName = generateUniqueName();
+        createTableAndEnableCDC(conn, tableName);
+
+        assertEquals("Post split partition update failures should be 0 initially",
+                0, METRICS_SOURCE.getPostSplitPartitionUpdateFailureCount());
+        assertEquals("Post merge partition update failures should be 0 initially",
+                0, METRICS_SOURCE.getPostMergePartitionUpdateFailureCount());
+
+        // Perform a split operation - this will fail 24 times before succeeding
+        TestUtil.splitTable(conn, tableName, Bytes.toBytes("m"));
+
+        // Verify split metric is 24
+        assertEquals("Post split partition update failures should be 15 after retries",
+                24, METRICS_SOURCE.getPostSplitPartitionUpdateFailureCount());
+
+        List<HRegionLocation> regions = TestUtil.getAllTableRegions(conn, tableName);
+
+        // Perform a merge operation - this will fail 15 times before succeeding
+        TestUtil.mergeTableRegions(conn, tableName, regions.stream()
+                .map(HRegionLocation::getRegion)
+                .map(RegionInfo::getEncodedName)
+                .collect(Collectors.toList()));
+
+        // Verify merge metric is 15
+        assertEquals("Post merge partition update failures should be 15 after retries",
+                15, METRICS_SOURCE.getPostMergePartitionUpdateFailureCount());
+
+        ResultSet rs = conn.createStatement().executeQuery(
+                "SELECT * FROM SYSTEM.CDC_STREAM WHERE TABLE_NAME='" + tableName + "'");
+        List<PartitionMetadata> mergedDaughter = new ArrayList<>();
+        List<PartitionMetadata> splitParents = new ArrayList<>();
+        while (rs.next()) {
+            PartitionMetadata pm = new PartitionMetadata(rs);
+            if (pm.startKey == null && pm.endKey == null && pm.endTime == 0) {
+                mergedDaughter.add(pm);
+            }
+            if (pm.startKey != null || pm.endKey != null) {
+                splitParents.add(pm);
+            }
+        }
+        assertEquals(2, mergedDaughter.size());
+        assertEquals(2, splitParents.size());
+        assertEquals(mergedDaughter.get(0).startTime, mergedDaughter.get(1).startTime);
+        assertEquals(mergedDaughter.get(0).endTime, mergedDaughter.get(1).endTime);
+        assertEquals(mergedDaughter.get(0).partitionId, mergedDaughter.get(1).partitionId);
+        assertTrue(mergedDaughter.stream().anyMatch(
+                d -> Objects.equals(d.parentPartitionId, splitParents.get(0).partitionId)));
+        assertTrue(mergedDaughter.stream().anyMatch(
+                d -> Objects.equals(d.parentPartitionId, splitParents.get(1).partitionId)));
+        for (PartitionMetadata splitDaughter : splitParents) {
+            Assert.assertEquals(mergedDaughter.get(0).startTime, splitDaughter.endTime);
+        }
+    }
+
+    private void createTableAndEnableCDC(Connection conn, String tableName) throws Exception {
```

Comment:
Should we move the util methods to a common place and re-use in both test classes?

Reply:
Let me however refactor by creating common utility.

Reply:
Yeah like `PartitionMetadata` and `createTableAndEnableCDC` etc can be shared

Reply:
Done